Aihe: [Intel-gfx] [RFC/PATCH xf86-video-intel] sna: Let modestting + glamor handle gen9+
Lähettäjä: Hans de Goede <hdegoede@redhat.com>
Päiväys: 29.02.2016 16:47
Vastaanottaja: intel-gfx@lists.freedesktop.org
CC: Hans de Goede <hdegoede@redhat.com>, Rob Clark <rclark@redhat.com>, Adam Jackson <ajax@redhat.com>

sna has no meaningfull accel for gen9+, this causes problems with i.e.
apps using XVideo since the sprite XVideo support does not work well
for many apps.

Therefor it is better to just let the xserver fall back to modesetting +
glamor. This is implemented by returning FALSE from the probe methods,
just like how nouveau handles falling back to modesetting for newer cards.

BugLink: https://bugzilla.redhat.com/show_bug.cgi?id=1305369
Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 src/intel_module.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/intel_module.c b/src/intel_module.c
index 60835b9..5979cb9 100644
--- a/src/intel_module.c
+++ b/src/intel_module.c
@@ -571,6 +571,13 @@ intel_scrn_create(DriverPtr		driver,
 			return FALSE;
 	}
 
+	/*
+	 * We've no accel support for these, so let modesetting + glamor
+	 * handle them.
+	 */
+	if ((unsigned)((struct intel_device_info *)match_data)->gen >= 0110)
+		return FALSE;
+
 	scrn = xf86AllocateScreen(driver, flags);
 	if (scrn == NULL)
 		return FALSE;
-- 2.7.2 _______________________________________________ Intel-gfx mailing list Intel-gfx@lists.freedesktop.org https://lists.freedesktop.org/mailman/listinfo/intel-gfx

